src/profile-logic/marker-schema.js

```diff
@@ -394,8 +404,118 @@ export function formatFromMarkerSchema(
       return formatPercent(value);
     default:
       console.error(
-        `A marker schema of type "${markerType}" had an unknown format "${(format: empty)}"`
+        `A marker schema of type "${markerType}" had an unknown format ${JSON.stringify(
+          format
+        )}`
       );
       return value;
   }
 }
+
+export function isDOMRenderingFormat(format: MarkerFormatType): boolean {
+  return format === 'url' || typeof format === 'object' || format === 'list';
+}
+
+function _formatDOMFromMarkerSchema(
+  markerType: string,
+  format: MarkerFormatType,
+  value: any,
+  // add elements that allow interaction (like clickable urls)
+  supportsInteraction: boolean
+) {
+  if (!isDOMRenderingFormat(format)) {
+    return <>{formatFromMarkerSchema(markerType, format, value)}</>;
+  }
+  if (typeof format === 'object') {
+    switch (format.type) {
+      case 'table': {
+        const { columns } = format;
+        if (!(value instanceof Array)) {
+          throw new Error('Expected an array for table type');
+        }
+        const hasHeader = columns.some((column) => column.label);
+        return (
+          <table>
```

Comment:
We'll probably want a class here, such as `tooltipMarkerTable`, and style it a bit, but I'd suggest this to be in a follow-up PR.

Reply:
No, I can add classes right away.

But you didn't find the real issue here :P

I didn't use the `<table>` element correctly: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table

Reply:
Well, if you're talking about thead/tbody, they're optional :)

Reply:
I get warnings in the web developer console nonetheless

Reply:
that's very weird, what are your warnings?

Reply:
<img width="839" alt="image" src="https://user-images.githubusercontent.com/490655/192001955-e24e8c63-26fe-45f0-9e16-84366dde850b.png">

Reply:
Appears only in Firefox, not in the embedded Chrome:

Warning: validateDOMNesting(...): <tr> cannot appear as a child of <table>. Add a <tbody>, <thead> or <tfoot> to your code to match the DOM tree generated by the browser.
tr
table
TooltipDetail@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:96968:23
div
TooltipDetails@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:96985:24
div
MarkerTooltipContents@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:95915:5
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
div
aside
MarkerSidebarImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:84930:1
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
div
s@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:48335:1317
div
t@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:48335:2968
DetailsContainerImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:64822:30
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
div
s@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:48335:1317
div
t@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:48335:2968
div
s@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:48335:1317
div
t@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:48335:2968
div
div
KeyboardShortcut@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:65805:5
ProfileViewerImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:68991:1
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
AppViewRouterImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:63782:1
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
UrlManagerImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:70082:5
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
div
DragAndDropImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:64914:5
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
LocalizationProvider@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:4145:142
AppLocalizationProviderImpl@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:63665:1
ConnectFunction@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46776:121
Provider@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:46489:18
ErrorBoundary@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:65135:5
Root@http://localhost:59300/main.e6904fb00509d3934052.bundle.js:69148:1 react_devtools_backend.js:4026:25

Reply:
I'm using the react dev tools, it seems that they are complaining, but I'm going to check whether this is really valid or a bug in the react dev tools.

Reply:
I looked it up and there's an explanation: the browser will insert <tbody> <thead> automatically to the DOM. As a result, when React will want to update, there may be problems because the DOM tree will be different than react's virtual dom tree.

I don't know if that's still a problem in React, but there's no harm adding them.

Reply:
See this issue: https://github.com/facebook/react/issues/5652 and the comments from waldreiter as well as sophiebits.